src/OpenTelemetry.Api/Baggage/BaggageManagerBase.cs

```diff
@@ -0,0 +1,35 @@
+// <copyright file="BaggageManagerBase.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using OpenTelemetry.Api.Context.Propagation;
+
+namespace OpenTelemetry.Api.Baggage
+{
+    public abstract class BaggageManagerBase
+    {
+        public abstract string GetEntry(BaggageContext context, string key);
+
+        public abstract BaggageContext SetEntry(BaggageContext context, string key, string value);
```

Comment:
is it going to create a new BaggageContext or update one passed in the argument? If former, it seems inefficient. If latter, this should be changes to the method on BaggageContext